### PR TITLE
Single case for CorruptThenRebuild and CorreuptThenRepair

### PR DESCRIPTION
**Pull request by harelassayag**
commit 1709d9f57aae74c262bdd74da5babf1278f2d533
Author: Harel Assayag <harel@scylladb.com>
Date:   Thu Feb 8 16:54:59 2018 +0200

    edit yamls

commit e430a919e69f8bffcc2105f85cc67163c620ef33
Author: Harel Assayag <harel@scylladb.com>
Date:   Wed Feb 7 15:08:50 2018 +0200

    added wait for prepare write to finish

commit 2a479b4c926b167b85c0759d4111dda79a512e81
Author: Harel Assayag <harel@scylladb.com>
Date:   Wed Feb 7 11:41:38 2018 +0200

    added tests

### CODE REVIEW

File: corrupt_then_rebuild_test.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+
+class CorruptThenRebuildTest(ClusterTester):
+    """
+    :avocado: enable
+    """
+    def test_corrupt_then_rebuild_nodes(self):
+
+        # run a prepare write workload
+        stress_queue = self.run_stress_thread(stress_cmd=self.params.get('stress_write_cmd'),
```
Comment (roydahan) [inline corrupt_then_rebuild_test.py]: I would like to use a faster way to populate data and use at least 500GB in the test.

File: corrupt_then_rebuild_test.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                                              stress_num=1,
+                                              keyspace_num=1)
+        #wait for prepare write to finish
+        self.db_cluster.wait_total_space_used_per_node()
+
```
Comment (roydahan) [inline corrupt_then_rebuild_test.py]: I would like it to have a read workload as well before starting rebuild.

File: corrupt_then_rebuild_test.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        # run rebuild
+        nm = nemesis.CorruptThenRebuildMonkey(self.db_cluster, self.loaders, self.monitors, None)
```
Comment (roydahan) [inline corrupt_then_rebuild_test.py]: please use better name than nm

File: destroy_data_then_repair_test.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+from avocado import main
+from sdcm.tester import ClusterTester
+from sdcm import nemesis
+
+
+class CorruptThenRepair(ClusterTester):
+    """
+    :avocado: enable
+    """
+    def test_destroy_data_then_repair_test_nodes(self):
```
Comment (roydahan) [inline destroy_data_then_repair_test.py]: Same as rebuild comments.

File: tests/destroy-data-then-repair.yaml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+ip_ssh_connections: 'private'
+experimental: 'true'
+round_robin: 'true'
+
+
+backends: !mux
+    aws: !mux
+        cluster_backend: 'aws'
+        instance_type_loader: 'c4.4xlarge'
+        instance_type_monitor: 'c4.2xlarge'
```
Comment (roydahan) [inline tests/destroy-data-then-repair.yaml]: change to t2.small

File: tests/destroy-data-then-repair.yaml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+space_node_threshold: 5000
+ip_ssh_connections: 'private'
+experimental: 'true'
+round_robin: 'true'
+
+
+backends: !mux
+    aws: !mux
+        cluster_backend: 'aws'
+        instance_type_loader: 'c4.4xlarge'
```
Comment (roydahan) [inline tests/destroy-data-then-repair.yaml]: change to c4.large

File: tests/destroy-data-then-repair.yaml [hunk 7]
```diff
@@ -0,0 +1,38 @@
+test_duration: 10080
+stress_write_cmd: "cassandra-stress write cl=QUORUM n=10000 -schema 'replication(factor=3) compaction(strategy=LeveledCompactionStrategy)' -port jmx=6868 -mode cql3 native -rate threads=500 -col 'size=FIXED(1024) n=FIXED(1)' -pop seq=1..300300300"
+stress_read_cmd:  "cassandra-stress read cl=ONE n=10000 -mode cql3 native -rate threads=10 -col 'size=FIXED(1024) n=FIXED(1)' -pop seq=1..1100200300 -port jmx=6868 -log interval=5"
+n_db_nodes: 3
+n_loaders: 4
```
Comment (roydahan) [inline tests/destroy-data-then-repair.yaml]: n_loaders is depended on the way we choose to load the data, in the current case there is no need for 4.
but it should change anyway.

File: tests/destroy-data-then-repair.yaml [hunk 8]
```diff
@@ -0,0 +1,38 @@
+test_duration: 10080
+stress_write_cmd: "cassandra-stress write cl=QUORUM n=10000 -schema 'replication(factor=3) compaction(strategy=LeveledCompactionStrategy)' -port jmx=6868 -mode cql3 native -rate threads=500 -col 'size=FIXED(1024) n=FIXED(1)' -pop seq=1..300300300"
+stress_read_cmd:  "cassandra-stress read cl=ONE n=10000 -mode cql3 native -rate threads=10 -col 'size=FIXED(1024) n=FIXED(1)' -pop seq=1..1100200300 -port jmx=6868 -log interval=5"
+n_db_nodes: 3
```
Comment (roydahan) [inline tests/destroy-data-then-repair.yaml]: Change to 4 - we aren't going to use refresh.

File: tests/destroy-data-then-repair.yaml [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+n_loaders: 4
+n_monitor_nodes: 1
+nemesis_interval: 15
+user_prefix: 'cases-repair-VERSION'
+failure_post_behavior: destroy
+space_node_threshold: 5000
```
Comment (roydahan) [inline tests/destroy-data-then-repair.yaml]: That's too small.

File: tests/destroy-data-then-repair.yaml [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+            ami_loader_user: 'centos'
+            ami_monitor_user: 'centos'
+
+databases: !mux
+    cassandra:
+        db_type: cassandra
+        instance_type_db: 'm3.large'
+    scylla:
+        db_type: scylla
+        instance_type_db: 'i2.4xlarge'
```
Comment (roydahan) [inline tests/destroy-data-then-repair.yaml]: change to i3.2xlarge

File: tests/corrupt-then-rebuild.yaml [hunk 11]
```diff
@@ -0,0 +1,38 @@
+test_duration: 10080
```
Comment (roydahan) [inline tests/corrupt-then-rebuild.yaml]: Same comments as repair yaml

File: tests/destroy-data-then-repair.yaml [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+n_loaders: 4
+n_monitor_nodes: 1
+nemesis_interval: 15
+user_prefix: 'cases-repair-VERSION'
+failure_post_behavior: destroy
+space_node_threshold: 5000
+ip_ssh_connections: 'private'
+experimental: 'true'
+round_robin: 'true'
+
```
Comment (roydahan) [inline tests/destroy-data-then-repair.yaml]: Add a provision_allocation for spot_low_price

File: corrupt_then_rebuild_test.py (hunk 1 above)
Comment (amoskong) [reply corrupt_then_rebuild_test.py]: multiple loaders/stress with different population seq range?

**Pull request closed without merging**